emcmake.bat

```diff
@@ -10,9 +10,6 @@
 
 :: All env. vars specified in this file are to be local only to this script.
 @setlocal
-:: -E will not ignore _PYTHON_SYSCONFIGDATA_NAME an internal
-:: of cpython used in cross compilation via setup.py.
-@set _PYTHON_SYSCONFIGDATA_NAME=
```

Comment:
These files are auto-generated.  See the comment at the top of the file.

Reply:
Ahh, I suspected that but missed the comment. That begs the question, should it change for all of them? The GCC wrappers (probably!) aren't going to be calling Python to build extension modules, while CMake does. Thought was this just added due to standard practice with the `-E`, or was it really needed? Not sure having sysconfig report the cross-compiling Python is likely to affect things that aren't building extensions.

Reply:
(I'm traveling at a conference so slow/limited currently)

Reply:
Not really sure why _PYTHON_SYSCONFIGDATA_NAME was added explicitly here.   Perhaps @pmp-p remembers?  

I think we should probably just remove it from all of them, unless there is a real good reason not to.

Reply:
Flagging @katietz who is likely interested in this discussion.

Reply:
I'm still not clear what would break if we simply stopped stripping _PYTHON_SYSCONFIGDATA_NAME completely?  

@henryii if you want to completely avoid emcmake, why can't you just replace it with `cmake -DCMAKE_CROSSCOMPILING_EMULATOR=... -DCMAKE_TOOLCHAIN_FILE=...` ?   It seems like you must have control of the cmake command line if you are already change `cmake ..` to `emcmake cmake ...`?

Reply:
> _PYTHON_SYSCONFIGDATA_NAME will be reused in any python subprocess spawned by emcc if found in env so it must always be removed or emsdk will use the python cross compiler sysconfig for compiling ( and often fail ).

Can you explain the problem a little more?

If somebody sets `_PYTHON_SYSCONFIGDATA_NAME` during compilation aren't they saying that they want to use the sysconf data? 

When does emcc "use the sysconfig for compiling" ?   Are you talking about `emcmake` and `emconfigure` maybe?

Reply:
> why can't you just replace it with

That's what I'm working on. I just need to figure out how to get the proper contents of `-DCMAKE_CROSSCOMPILING_EMULATOR` (that is, I need to import `tools` I think). Though this is only a fix for pyodide-build; other emcmake users won't benefit.

> during compilation aren't they saying that they want to use the sysconf data?

Yes, exactly, but all the wrapper scripts strip it out and keep any subprocesses from accessing it. So `_PYTHON_SYSCONFIGDATA_NAME=... emcmake cmake ...` will not cross compile Python modules, since cmake can't see `_PYTHON_SYSCONFIGDATA_NAME`.

I don't know about emcc - does `_PYTHON_SYSCONFIGDATA_NAME` break something it is doing (as it's a lengthy Python script with quite a few imports), or does it break something it calls? If the later, then it could just strip out the variable when it does the calling. If the former, then it needs to be removed before it starts up. But that's something that's very much specific to that script, and it's something it could even restore after starting up, and isn't correct to strip from all wrappers.

Reply:
I can try removing it and seeing what breaks. :)

Reply:
Actually, even it it breaks `emcc` itself, that's fixable, as `sysconfig` isn't in the default packages loaded by site.py. You can just do:

```
import os
if "_PYTHON_SYSCONFIGDATA_NAME" in os.eviron:
    old = os.environ.pop("_PYTHON_SYSCONFIGDATA_NAME")
    import sysconfig
    os.eviron["_PYTHON_SYSCONFIGDATA_NAME"] = old
```

at the top. The original sysconfigdata module will be loaded.

Reply:
That sounds like a reasonable hack yes.

I would like to be able to write a test for this.. is there some way we can make emscripten fail given a bad _PYTHON_SYSCONFIGDATA_NAME?  What kind of failure would I expect?

Reply:
Presumably we would set `_PYTHON_SYSCONFIGDATA_NAME` to something like `"emscripten_3_1_32_wasm32"`.

Reply:
FYI, you can set `_PYTHON_SYSCONFIGDATA_NAME=nothing-real` and see exactly where sysconfig is being imported. If it doesn't fail, sysconfig is never imported.